influx/templates.go

```diff
@@ -1,40 +1,106 @@
 package influx
 
 import (
+	"sort"
+	"strconv"
 	"strings"
+	"time"
 
 	"github.com/influxdata/chronograf"
 )
 
-// TemplateReplace replaces templates with values within the query string
-func TemplateReplace(query string, templates chronograf.TemplateVars) string {
-	tvarsByPrecedence := make(map[uint]chronograf.TemplateVars, len(templates))
-	maxPrecedence := uint(0)
-	for _, tmp := range templates {
-		precedence := tmp.Precedence()
-		if precedence > maxPrecedence {
-			maxPrecedence = precedence
-		}
-		tvarsByPrecedence[precedence] = append(tvarsByPrecedence[precedence], tmp)
-	}
+func SortTemplates(ts []chronograf.TemplateVar) []chronograf.TemplateVar {
+	sort.Slice(ts, func(i, j int) bool {
+		if len(ts[i].Values) != len(ts[j].Values) {
+			return len(ts[i].Values) < len(ts[j].Values)
+		}
 
-	replaced := query
-	for prc := uint(0); prc <= maxPrecedence; prc++ {
-		replacements := []string{}
+		if len(ts[i].Values) == 0 {
+			return i < j
+		}
 
-		for _, v := range tvarsByPrecedence[prc] {
-			if evar, ok := v.(chronograf.ExecutableVar); ok {
-				evar.Exec(replaced)
+		for k := range ts[i].Values {
+			if ts[i].Values[k].Type != ts[j].Values[k].Type {
+				return ts[i].Values[k].Type < ts[j].Values[k].Type
 			}
-			newVal := v.String()
-			if newVal != "" {
-				replacements = append(replacements, v.Name(), newVal)
+			if ts[i].Values[k].Value != ts[j].Values[k].Value {
+				return ts[i].Values[k].Value < ts[j].Values[k].Value
 			}
 		}
+		return i < j
+	})
+	return ts
+}
+
+// RenderTemplate converts the template variable into a correct InfluxQL string based
+// on its type
+func RenderTemplate(query string, t chronograf.TemplateVar, now time.Time) (string, error) {
```

Comment:
A little documentation of the `why` would awesome here. The code makes sense, but a little high level of the specific goals would be useful. Maybe, I wrote this on my first read. As I read it more, I feel less and less strongly about this.